src/threadpool-object.h

```diff
@@ -20,6 +20,12 @@
 
 /* Windows headers */
 #if PTHREADPOOL_USE_EVENT
+#ifndef NOMINMAX
```

Comment:
NOMINMAX is unnecessary: some SDKs ignore it when building in C mode, and pthreadpool has undefines min macro anyway before defining the corresponding function.

Reply:
It can keep the header clean, if downstream uses this header.

Reply:
This is a private header, it can't be used downstream

Reply:
2 experiments:
1. stdlib.h introduced a `min`, probably by itself since winsdk is correctly controlled by `NOMINMAX`
<img width="487" alt="image" src="https://user-images.githubusercontent.com/5203025/101687972-3dc16a00-3aa6-11eb-9c27-c0d4789cb9df.png">
2. `WIN32_LEAN_AND_MEAN` is not enough:
<img width="426" alt="image" src="https://user-images.githubusercontent.com/5203025/101688068-6b0e1800-3aa6-11eb-9237-db6df524615f.png">

Reply:
Could you check those 2 experiment I just did?
If you still think `NOMINMAX` should not be added, I can drop them.

Reply:
i.e. you probably have 3 `min` defined there:
1. winsdk macro
2. msvc stdlib.h macro
3. your version
NOMINMAX get rid of the first one.
> some SDKs ignore it when building in C mode
I also wonder if this is actually the result of stdlib.

Reply:
Even if `min` macro comes from `stdlib.h`, there's no reason to define `NOMINMAX`, as the end result is the same (`min` macro is defined)

Reply:
Yes, I think it would be more clear to leave out `NOMINMAX` completely.